spring-kafka/src/test/java/org/springframework/kafka/core/DefaultKafkaConsumerFactoryTests.java

```diff
@@ -59,18 +65,149 @@
+			protected KafkaConsumer<String, String> createKafkaConsumer(Map<String, Object> configProps) {
+				assertThat(configProps.get(ConsumerConfig.CLIENT_ID_CONFIG)).isEqualTo("overridden");
 				return null;
 			}
+		}.createConsumer(null, "overridden", null, null);
+	}
+
+	@Test
+	public void testClientIdSuffixAndPrefixOnDefault() {
+		Map<String, Object> originalConfig = Collections.singletonMap(ConsumerConfig.CLIENT_ID_CONFIG, "original");
+		new DefaultKafkaConsumerFactory<String, String>(originalConfig) {
+			protected KafkaConsumer<String, String> createKafkaConsumer(Map<String, Object> configProps) {
+				assertThat(configProps.get(ConsumerConfig.CLIENT_ID_CONFIG)).isEqualTo("overridden-1");
+				return null;
+			}
+		}.createConsumer(null, "overridden", "-1", null);
+	}
+
+	@Test
+	public void testClientIdSuffixAndPrefixOnPropertyOverride() {
+		Map<String, Object> originalConfig = Collections.singletonMap(ConsumerConfig.CLIENT_ID_CONFIG, "original");
+		Properties overrides = new Properties();
+		overrides.setProperty(ConsumerConfig.CLIENT_ID_CONFIG, "property-overridden");
+		new DefaultKafkaConsumerFactory<String, String>(originalConfig) {
+			protected KafkaConsumer<String, String> createKafkaConsumer(Map<String, Object> configProps) {
+				assertThat(configProps.get(ConsumerConfig.CLIENT_ID_CONFIG)).isEqualTo("overridden-1");
+				return null;
+			}
+		}.createConsumer(null, "overridden", "-1", overrides);
+	}
 
-		};
-		factory.createConsumer("-1");
+
+	@Test
+	public void testGroupIdOnDefault() {
+		Map<String, Object> originalConfig = Collections.singletonMap(ConsumerConfig.GROUP_ID_CONFIG, "original");
+		new DefaultKafkaConsumerFactory<String, String>(originalConfig) {
+			protected KafkaConsumer<String, String> createKafkaConsumer(Map<String, Object> configProps) {
+				assertThat(configProps.get(ConsumerConfig.GROUP_ID_CONFIG)).isEqualTo("overridden");
+				return null;
+			}
+		}.createConsumer("overridden", null, null, null);
+	}
+
+	@Test
+	public void testGroupIdWithoutDefault() {
+		new DefaultKafkaConsumerFactory<String, String>(Collections.emptyMap()) {
+			protected KafkaConsumer<String, String> createKafkaConsumer(Map<String, Object> configProps) {
+				assertThat(configProps.get(ConsumerConfig.GROUP_ID_CONFIG)).isEqualTo("overridden");
+				return null;
+			}
+		}.createConsumer("overridden", null, null, null);
+	}
+
+	@Test
+	public void testGroupIdOnPropertyOverride() {
+		Map<String, Object> originalConfig = Collections.singletonMap(ConsumerConfig.GROUP_ID_CONFIG, "original");
+		Properties overrides = new Properties();
+		overrides.setProperty(ConsumerConfig.GROUP_ID_CONFIG, "property-overridden");
+		new DefaultKafkaConsumerFactory<String, String>(originalConfig) {
+			protected KafkaConsumer<String, String> createKafkaConsumer(Map<String, Object> configProps) {
+				assertThat(configProps.get(ConsumerConfig.GROUP_ID_CONFIG)).isEqualTo("overridden");
```

Comment:
Same test code flow concern in all your new tests.

Reply:
agreed